sw/airborne/modules/system_identification/pprz_chirp.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) Joost Meulenbeld
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/system_identification/pprz_chirp.c"
+ * @author Joost Meulenbeld
+ * Mathematical implementation of the chirp
+ */
+#include "pprz_chirp.h"
+
+
+void chirp_init(struct chirp_t *chirp, float f0_hz, float f1_hz, float length_s, float current_time_s,
+                bool exponential_chirp, bool fade_in) {
+    chirp->f0_hz = f0_hz;
+    chirp->f1_hz = f1_hz;
+
+    if (length_s > 0) {
+        chirp->length_s = length_s;
+    } else {
+        fprintf(stderr, "chirp_t->length_s must be a positive value");
+        exit(1);
```

Comment:
I added a check to avoid division by zero in a later portion of the code with total_length_s, but I am not sure on the conventions used to raise errors